lib/statsd.rb

```diff
@@ -14,14 +14,77 @@
 #   statsd = Statsd.new('localhost').tap{|sd| sd.namespace = 'account'}
 #   statsd.increment 'activate'
 class Statsd
-  class RubyUdpClient
-    attr_reader :key, :sock
+  class BufferedUDPClient < UDPClient
+    @max_packet_size = 511
+    (class << self; attr_accessor :max_packet_size; end)
 
-    def initialize(address, port, key = nil)
+    def initialize(address, port)
+      super(address, port)
+      @buffer = String.new
+    end
+
+    def flush!
+      return unless @buffer.bytesize > 0
+      sock.write(@buffer)
+      @buffer.clear
+    rescue => boom
+      nil
+    end
+
+    def send(msg)
+      flush! if @buffer.bytesize + msg.bytesize > BufferedUDPClient.max_packet_size
+      @buffer << msg
+      @buffer << "\n".freeze
+      nil
```

Comment:
Right. I agree. In practice this won't matter, because this is based around MTUs and affected by the network's congestion, but I agree that the semantics should be right. Let me fix it.